src/auth/utils.ts

```diff
@@ -556,3 +557,55 @@ export class AuthNode implements TreeNode<Auth> {
         }
     }
 }
+
+/**
+ * Class to get info about the user + use of this extension
+ *
+ * Why is this extension in this file?
+ * - Due to circular dependency issues since this class needs to use the {@link Auth}
+ *   instance. If we can find a better spot and not run in to the isssue this should be moved.
+ *
+ * Keywords for searchability:
+ * - new user
+ * - first time
+ */
+export class ExtensionUse {
+    public readonly isExtensionFirstUseKey = 'isExtensionFirstUse'
+
+    // The result of if is first use for the remainder of the extension session.
+    // This will reset on next startup.
+    private isFirstUseCurrentSession: boolean | undefined
+
+    isFirstUse(
+        state: vscode.Memento = globals.context.globalState,
+        hasExistingConnections = () => Auth.instance.hasConnections
+    ): boolean {
+        if (this.isFirstUseCurrentSession !== undefined) {
+            return this.isFirstUseCurrentSession
+        }
+
+        this.isFirstUseCurrentSession = state.get(this.isExtensionFirstUseKey)
+
+        if (this.isFirstUseCurrentSession === undefined) {
+            // The variable in the store is not defined yet, fallback to checking if they have existing connections.
+            this.isFirstUseCurrentSession = !hasExistingConnections()
+
+            getLogger().info(
+                `isFirstUse: State not found, marking them as '${
+                    this.isFirstUseCurrentSession ? '' : 'NOT '
+                }first use' since they 'did ${this.isFirstUseCurrentSession ? 'NOT ' : ''}have existing connections'.`
```

Comment:
For NOT first-time use this should be logged at verbose or debug log-level. Else it will always appear in the logs for most users.

Reply:
This scenario only triggers when the global state variable is `undefined`, when the function has never run. After this function runs it will set the global state variable to `false`.

So this only runs the first ever time this function is run for a user, then never again since the value will either be `true` or `false. Did you still not want to log it with `info` for all first time users?

Reply:
ok, info is fine then.